app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -211,6 +211,106 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
           }
         }
       }
+    },
+
+    ecommerceHelperFunctions: {
+      clearEcommerceObject: function() {
+        window.GOVUK.analyticsGa4.core.sendData({ search_results: { ecommerce: null } })
+      },
+
+      getIndex: function(data) {
```

Comment:
I’ve added a check and I’ve also done a very small refactor to take the parameters out of the object. I think putting parameters within an object is good if there are many parameters to pass, but in this case I think it might be more readable without it.